src/golang/lib/dynamic/engine.go

```diff
@@ -318,5 +333,44 @@ func ResyncClusterState(
 	// inconsistent state between the database and terraform. In this case, we resync the state by
 	// deleting the cluster and updating the database state to be Terminated.
 	log.Error("Dynamic k8s cluster might be in an inconsistent state. Resolving state by deleting the cluster...")
-	return DeleteDynamicEngine(ctx, engineIntegration, integrationRepo, db)
+	return DeleteDynamicEngine(ctx, true, engineIntegration, integrationRepo, db)
+}
+
+// safeToDeleteCluster checks whether there are pods in the aqueduct namespace of the dynamic k8s
+// cluster that are in ContainerCreating or Running status. If so, it returns false. Otherwise it
+// returns true.
+func safeToDeleteCluster(
+	ctx context.Context,
+	engineIntegration *models.Integration,
+) (bool, error) {
+	useSameCluster, err := strconv.ParseBool(engineIntegration.Config[shared.K8sUseSameClusterKey])
+	if err != nil {
+		return false, errors.Wrap(err, "Error parsing use_same_cluster flag")
+	}
+
+	k8sClient, err := k8s.CreateK8sClient(engineIntegration.Config[shared.K8sKubeconfigPathKey], useSameCluster)
```

Comment:
For better reusability, should the client be handled and passed around by caller?

Reply:
I pushed this helper down to `lib/k8s/dynamic.go` and am creating the client within the helper. This way I think the scope of this function is clearer.